### PR TITLE
Fixes About Task lists article

### LINKED ISSUE #9317: Improved Task list not available on GHES
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://github.com/github/docs/edit/main/content/issues/tracking-your-work-with-issues/about-task-lists.md

### What changes are you suggesting?

The information about the task list suggest that some of the new features (like automatic extension to full name, reordering tasks) are as well available in GHES (e.g. in 3.0: https://docs.github.com/en/enterprise-server@3.0/issues/tracking-your-work-with-issues/about-task-lists), while they are only available in beta on gtihub.com.

The documentation in my opinion should be updated to reflect that this things are currently only available on github.com and on GHES still the old plain markdown task list exists.

### Additional information

_No response_

---

### Content design plan to fix this issue

A couple of small changes to the [About task lists](https://docs.github.com/en/enterprise-server@3.1/issues/tracking-your-work-with-issues/about-task-lists) article are needed:

1. [Reordering tasks](https://docs.github.com/en/enterprise-server@3.0/issues/tracking-your-work-with-issues/about-task-lists#reordering-tasks): Create an image or animated GIF showing how to reorder tasks on GitHub Enterprise Server, and show that in place of the current animated GIF (which shows the autoexpanding issue URLs, as you point out).
2. [Navigating tracked issues](https://docs.github.com/en/enterprise-server@3.0/issues/tracking-your-work-with-issues/about-task-lists#navigating-tracked-issues): Version this section for GitHub.com only.

### PR DESCRIPTION
**Pull request by thispsj**
### Why:

Closes #9317 

### What's being changed:

Changes suggested by [felicitymay](https://github.com/felicitymay) applied.

1) Versions "Navigating Issues" section to GitHub.com only
2) Added a new GIF for GHES wherein issue and pull request URLs aren't auto expanded.

### Check off the following:

- [x] I have reviewed my changes in staging (look for the latest deployment event in your pull request's timeline, then click **View deployment**).
- [X] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/CONTRIBUTING.md#self-review).

### CODE REVIEW

**Review by mchammer01 — COMMENTED** [review]
@thispsj 👋🏻 - these changes LGTM, thanks for improving our docs ✨ 
I made some minor suggestions relating to our content model (ALT text) and about simplifying a versioning Liquid statement that I'm going to apply.
There are two outstanding issues re: the GIF file that I haven't been able to help with:

- The new GHES image needs to be placed in a different directory as it's GHES-only: `assets/images/help/writing/task-list-reordered-ghes.gif` -> `assets/images/enterprise/writing/task-list-reordered-ghes.gif`. Can you do that? I wasn't able to get your branch and make these changes myself. You'll need to create a new `writing` subdirectory in `assets/images/enterprise/`, place your GIF there, and remove the image from the current directory it's in.

- The GIF file looks blurred in the [preview](https://docs-10528--fix-task-lists.herokuapp.com/en/enterprise-server@3.1/issues/tracking-your-work-with-issues/about-task-lists#reordering-tasks). Compare your GIF with the [GitHub.com version](https://docs-10528--fix-task-lists.herokuapp.com/en/issues/tracking-your-work-with-issues/about-task-lists#reordering-tasks). We have our own optimization algorithn, so perhaps you should try adding your unoptimized GIF, and we'll see how things look.

Hope this helps.

**Comment by thispsj** [comment]
> @thispsj 👋🏻 - these changes LGTM, thanks for improving our docs ✨ I made some minor suggestions relating to our content model (ALT text) and about simplifying a versioning Liquid statement that I'm going to apply. There are two outstanding issues re: the GIF file that I haven't been able to help with:
> 
> * The new GHES image needs to be placed in a different directory as it's GHES-only: `assets/images/help/writing/task-list-reordered-ghes.gif` -> `assets/images/enterprise/writing/task-list-reordered-ghes.gif`. Can you do that? I wasn't able to get your branch and make these changes myself. You'll need to create a new `writing` subdirectory in `assets/images/enterprise/`, place your GIF there, and remove the image from the current directory it's in.
> * The GIF file looks blurred in the [preview](https://docs-10528--fix-task-lists.herokuapp.com/en/enterprise-server@3.1/issues/tracking-your-work-with-issues/about-task-lists#reordering-tasks). Compare your GIF with the [GitHub.com version](https://docs-10528--fix-task-lists.herokuapp.com/en/issues/tracking-your-work-with-issues/about-task-lists#reordering-tasks). We have our own optimization algorithn, so perhaps you should try adding your unoptimized GIF, and we'll see how things look.
> 
> Hope this helps.

Yeah I will do the changes you suggested.

@mchammer01 What about the suggested changes ? Should I commit them ?

**Comment by ramyaparimi** [comment]
@thispsj Thank you for enabling the option ✨  Enabling maintainer edits will quicken the PR review and merge process greatly 🐇  Could you please double check for all other PRs created now and in the future before submitting them ✅? 

Thanks so much for your passion in contributing to our docs 💖

**Comment by mchammer01** [comment]
@thispsj 👋🏻 - thanks for returning to this PR and completing the work. I'm currently working to a tight deadline, so I've unassigned myself as I won't have time to review your latest updates. I'm sure someone else from the team will review this for you 🙂

**Comment by thispsj** [comment]
@mchammer01 no problem

File: content/issues/tracking-your-work-with-issues/about-task-lists.md
Comment (felicitymay) [inline content/issues/tracking-your-work-with-issues/about-task-lists.md:59]: https://docs-10528--fix-task-lists.herokuapp.com/en/enterprise-server@3.2/issues/tracking-your-work-with-issues/about-task-lists shows a broken image. Looks as if this is a tiny typo.

```suggestion
{% else %} ![Reordered task list](/assets/images/enterprise/writing/task-lists-reorder.gif) {% endif %}
```

**Comment by felicitymay** [comment]
Thanks for moving the GIF 💖 

I've fixed a small typo and updated from `main`. I'll check the preview and then 🤞🏻we'll be ready to merge your change once the repo is unfrozen 🙂

**Review by felicitymay — APPROVED** [review]
Thanks for the changes. This looks ready to merge when the repo is unfrozen.

**Pull request merged**
